lib/authn.js

```diff
@@ -0,0 +1,90 @@
+// @ts-check
+const config = require('../lib/config');
+const csrf = require('../lib/csrf');
+const sqldb = require('@prairielearn/postgres');
+
+const sql = sqldb.loadSqlEquiv(__filename);
+
+/**
+ * @typedef {Object} LoadUserOptions
+ * @property {boolean} [LoadUserOptions.pl_authn_cookie=true] - Create the cookie?
+ * @property {boolean} [LoadUserOptions.redirect=false] - Redirect after processing?
```

Comment:
No need for the `LoadUserOptions.` prefix in the property name, and type information for an object can't capture defaults, so the `=...` is unnecessary.

```suggestion
 * @property {boolean} [pl_authn_cookie] - Create the cookie?
 * @property {boolean} [redirect] - Redirect after processing?
```